Sources/StreamChat/Database/DatabaseContainer_Mock.swift

```diff
@@ -18,7 +18,7 @@ class DatabaseContainerMock: DatabaseContainer {
     @Atomic var resetEphemeralValues_called = false
     
     convenience init() {
-        try! self.init(kind: .inMemory)
+        try! self.init(kind: .onDisk(databaseFileURL: .newTemporaryFileURL()))
```

Comment:
Shouldn't we try-catch this instead to know what's going on? This would surely bring some rewriting of the inits 😅

Reply:
tl;dr: You're probably right but I think it's OK.

---

Good comment. I think one of the main reasons for the convenience init is to hide the fact it throws.

However, this `init` is used only in tests that don't test the actual DB functionality. I think it's OK to hide the throwing here. Even if I added some inner `do { } catch { }` and log the error nicely, I'd still have to add a `fatalError` because it's a non-failable initializer. This container init will fail only if the CD stack fails to init which in our setup is a programmer error.